numpy/core/src/umath/ufunc_object.c

```diff
@@ -575,12 +776,41 @@ get_ufunc_arguments(PyUFuncObject *ufunc,
     int nout = ufunc->nout;
     int nop = ufunc->nargs;
     PyObject *obj, *context;
-    PyObject *str_key_obj = NULL;
-    const char *ufunc_name = ufunc_get_name_cstr(ufunc);
-    int has_sig = 0;
-
     /*
-     * Initialize objects so caller knows when outputs and other optional
+     * Keyword definitions.
+     *
+     * Note: if any changes are made, one also needs to change
+     * the initalization of the outputs in the kwds loop.
+     */
+    static PyObject **kwnames[] = {
+        &npy_um_str_out,
+        &npy_um_str_where,
+        &npy_um_str_axes,
+        &npy_um_str_axis,
+        &npy_um_str_keepdims,
+        &npy_um_str_casting,
+        &npy_um_str_order,
+        &npy_um_str_dtype,
+        &npy_um_str_subok,
+        &npy_um_str_signature,
+        &npy_um_str_sig,
+        &npy_um_str_extobj,
+        NULL};                  /* sentinel */
+    typedef int converter(PyObject *, void *);
+    static converter *_convert_casting = NULL;
+    static converter *_convert_order = NULL;
+    static converter *_convert_dtype = NULL;
+    /*
+     * Some converters are from the Array_API, which are not availabe'
+     * on compile time, so we insert them on the first call.
+     */
+    if (_convert_casting == NULL) {
+        _convert_casting = (converter *)PyArray_CastingConverter;
+        _convert_order = (converter *)PyArray_OrderConverter;
+        _convert_dtype = (converter *)PyArray_DescrConverter2;
```

Comment:
Is this really worth it vs just passing these functions directly into the `parse_ufunc_keywords` call below?

Reply:
Duh, yes, of course that should work! Indeed, much clearer, especially with the keywords there too.